docs/user/security/fips-140-2.asciidoc

```diff
@@ -0,0 +1,62 @@
+[role="xpack"]
+[[xpack-security-fips-140-2]]
+=== FIPS 140-2
+
+experimental[] The Federal Information Processing Standard (FIPS) Publication 140-2, (FIPS PUB 140-2), 
+titled "Security Requirements for Cryptographic Modules" is a U.S. government computer security standard
+used to approve cryptographic modules.
+
+{kib} offers a FIPS 140-2 compliant mode and as such can run in a Node.js environment configured with a FIPS 
+140-2 compliant OpenSSL3 provider.
+
+To run {kib} in FIPS mode, your cluster must be have a Platinum or Enterprise license.
+
+[IMPORTANT]
+============================================================================
+The Node bundled with {kib} is not configured for FIPS 140-2. You must configure a FIPS 140-2 compliant OpenSSL3 
+provider. Please consult the Node.js documentation to learn how to configure your environment.
+============================================================================
+
+For {kib}, adherence to FIPS 140-2 is ensured by:
+
+* Using FIPS approved / NIST recommended cryptographic algorithms.
+
+* Delegating the implementation of these cryptographic algorithms to a NIST validated cryptographic module
+(available via Node.js configured with an OpenSSL3 provider).
+
+* Allowing the configuration of {kib} in a FIPS 140-2 compliant manner, as documented below. 
+
+==== Configuring {kib} for FIPS 140-2
```

Comment:
Do we need to do anything about https://github.com/elastic/kibana/blob/975eeed255d0413f57827d830f991cdf66bdf788/config/node.options#L14-L15

Reply:
Good question 😅 Im not sure how that conflicts in a FIPS setup. I can do some investigation.

Is there anyway the KB admin could override this currently?

Reply:
It can be removed, Kibana archives will have this at `$KIBANA_HOME/config/node.options`. or for deb/rpm `/etc/kibana/node.options`.

Reply:
I _think_ it's ok to leave. You can use multiple providers alongside the FIPS provider. In #183777 I ran the FIPS agent pipeline appending FIPS options to the existing `node.options` without issue. That said, not sure it's "FIPS compliant" and still worth double checking @kc13greiner.

Reply:
Ok, we should be able to leave it:

When I configure node with:
```
[provider_sect]
default = default_sect
# The fips section name should match the section name inside the
# included fipsmodule.cnf.
# fips = fips_sect
legacy = legacy_sect
```

It runs with the legacy provider (w/o FIPS)  and allows `md4`/`md5`

<img width="1446" alt="Screenshot 2024-06-11 at 5 25 10 PM" src="https://github.com/elastic/kibana/assets/21210601/b68b2557-a877-4e65-b087-f9aad4c4b95d">


If I include `fips` as a provider, unsupported algorithms are not allowed and it doesn't even mention legacy mode anymore

```
[provider_sect]
default = default_sect
# The fips section name should match the section name inside the
# included fipsmodule.cnf.
fips = fips_sect
legacy = legacy_sect
```
<img width="1806" alt="Screenshot 2024-06-11 at 5 26 56 PM" src="https://github.com/elastic/kibana/assets/21210601/e6333090-4080-4859-9849-85d3c4dd0357">